plugins/inputs/cloudwatch/README.md

```diff
@@ -0,0 +1,90 @@
+# Amazon CloudWatch Statistics Input
+
+This plugin will pull Metric Statistics from Amazon CloudWatch.
+
+### Amazon Authentication
+
+This plugin uses a credential chain for Authentication with the CloudWatch
+API endpoint. In the following order the plugin will attempt to authenticate.
+1. [IAMS Role](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html)
+2. [Environment Variables](https://github.com/aws/aws-sdk-go/wiki/configuring-sdk#environment-variables)
+3. [Shared Credentials](https://github.com/aws/aws-sdk-go/wiki/configuring-sdk#shared-credentials-file)
+
+### Configuration:
+
+```toml
+[[inputs.cloudwatch]]
+  ## Amazon Region
+  region = 'us-east-1'
+
+  ## Requested [CloudWatch aggregation Period](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch-new.html) (must be a multiple of 60s)
+  period = '1m'
+
+  ## Collection Delay (must account for metrics availability via CloudWatch API)
+  delay = '1m'
+
+  ## Recomended: use metric 'interval' that is a multiple of 'period' to avoid 
+  ## gaps or overlap in pulled data
+  interval = '1m'
+
+  ## Array of Metric Statistics to pull
+  [[inputs.cloudwatch.metrics]]
+    tag = 'p-example-latency'
+    statistics = ['Average', 'Maximum']
+    namespace = 'AWS/ELB'
+    name = 'Latency'
+    unit = 'Seconds'
+    [[inputs.cloudwatch.metrics.dimensions]]
```

Comment:
Dimensions are CloudWatch's mechanism of uniquely identifying a Metric for a certain resource (or subset of resources that report that Metric in that Namespace).
The Metrics and Dimensions vary per service and are defined per-service here:
http://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/CW_Support_For_AWS.html

Take ELB Namespace for example - http://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/elb-metricscollected.html

If you ask for Metric `Latency`, you will get aggregated Latency for *all* your ELBs.  If you want to filter it to certain subset of ELBs, you can filter further on Dimensions - in this case `LoadBalancerName` and/or  `AvailabilityZone`.

Dimensions are optional, but are usually required to filter to desired granularity.